akka-remote-tests/src/multi-jvm/scala/akka/remote/artery/UdpPortActor.scala

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.actor.Actor
+import akka.remote.RARP
+import akka.testkit.SocketUtil
+
+object UdpPortActor {
+  case object GetUdpPort
+}
+
+/**
+ * Used for exchanging free udp port between multi-jvm nodes
+ */
+class UdpPortActor extends Actor {
+  import UdpPortActor._
+
+  val port = SocketUtil.temporaryServerAddress(RARP(context.system).provider
+    .getDefaultAddress.host.get, udp = true).getPort
+
+  def receive = {
+    case GetUdpPort ⇒ sender() ! port
+  }
+}
```

Comment:
Why did it go in an actor instead of just calling the logic directly?

Reply:
It’s retrieved from other nodes, see def channel

Reply:
Ah! Thanks for clarifying.